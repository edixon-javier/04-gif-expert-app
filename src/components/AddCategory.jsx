import PropTypes from "prop-types";
import { useState } from "react";


export const AddCategory = ({ onNewValue }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    setInputValue('');
    onNewValue(inputValue.trim());
    // setValueInput((cat) => [inputValue, ...cat]);
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs..."
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};


AddCategory.propTypes = {
  onNewValue: PropTypes.func.isRequired,
};
