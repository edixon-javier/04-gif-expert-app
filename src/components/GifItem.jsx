import PropTypes from "prop-types";

export const GifItem = ({ url, title }) => {
  return (
    <>
      <div className="card">
        <img src={url} alt={title} />
      </div>
    </>
  );
};

//! npm install --save-dev prop-types
//! npm add prop-types para agregar esto, se debe instalar la dependencias
GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
