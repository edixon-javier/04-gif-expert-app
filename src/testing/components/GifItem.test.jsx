/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";
describe("evidence in the archive GifItem", () => {
  const url = `https://api.giphy.com/v1/gifs/search?api_key=brkkTG9o4yTFN77IMDY9wFQuePAhp08Z&q=goku&limit=16`;
  const title = "goku";

  test("Must match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Must display the image and url indicated. ", () => {
    render(<GifItem title={title} url={url} />);
    //! screen es importante para saber como se ve cuando se render, tener en cuenta la importacion
    // screen.debug();
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
    //? estas pruebas son iguales pero desdestrcuturando el img para sacas las propiedades.
    expect(screen.getByRole("img").src).toBe(url);
    expect(screen.getByRole("img").alt).toBe(title);
  });
});
