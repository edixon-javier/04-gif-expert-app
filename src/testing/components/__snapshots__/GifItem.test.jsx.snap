// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`evidence in the archive GifItem Must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="goku"
      src="https://api.giphy.com/v1/gifs/search?api_key=brkkTG9o4yTFN77IMDY9wFQuePAhp08Z&q=goku&limit=16"
    />
  </div>
</div>
`;
